src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,388 @@
+            ComEventSink.Advise<IVsCodeWindowEvents>(codeWindow, this);
+            SortOption = SortOption.Order;
+
+            _getModelQueue = new AsyncBatchingWorkQueue(
+                    DelayTimeSpan.Short,
+                    GetModelAsync,
+                    asyncListener,
+                    threadingContext.DisposalToken);
+
+            _updateModelQueue = new AsyncBatchingWorkQueue(
+                    DelayTimeSpan.NearImmediate,
+                    UpdateModelAsync,
+                    asyncListener,
+                    threadingContext.DisposalToken);
+
+            _highlightNodeQueue = new AsyncBatchingWorkQueue(
+                    DelayTimeSpan.NearImmediate,
+                    HightlightNodeAsync,
+                    asyncListener,
+                    threadingContext.DisposalToken);
+
+            codeWindow.GetPrimaryView(out var pTextViewPrimary);
+            StartTrackingView(pTextViewPrimary, out var textViewPrimary);
+
+            // Primary text view should always exist on window initialization unless an error is thrown.
+            if (textViewPrimary is null)
+                return;
+
+            codeWindow.GetSecondaryView(out var pTextViewSecondary);
+            if (pTextViewSecondary is not null)
+                StartTrackingView(pTextViewSecondary, out var _);
+
+            StartGetModelTask();
+        }
+
+        int IVsCodeWindowEvents.OnNewView(IVsTextView pView)
+        {
+            StartTrackingView(pView, out var _);
+            return VSConstants.S_OK;
+        }
+
+        private void StartTrackingView(IVsTextView pTextView, out IWpfTextView? wpfTextView)
+        {
+            ThreadingContext.ThrowIfNotOnUIThread();
+            wpfTextView = null;
+            if (pTextView != null)
+            {
+                wpfTextView = EditorAdaptersFactoryService.GetWpfTextView(pTextView);
+                if (wpfTextView != null)
+                {
+                    _trackedTextViews.Add(pTextView, wpfTextView);
+                    wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+                    wpfTextView.TextBuffer.Changed += TextBuffer_Changed;
+                }
+            }
+        }
+
+        int IVsCodeWindowEvents.OnCloseView(IVsTextView pView)
+        {
+            if (_trackedTextViews.TryGetValue(pView, out var view))
+            {
+                view.Caret.PositionChanged -= Caret_PositionChanged;
+                view.TextBuffer.Changed -= TextBuffer_Changed;
+
+                _trackedTextViews.Remove(pView);
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        private void TextBuffer_Changed(object sender, TextContentChangedEventArgs e)
+            => StartGetModelTask();
+
+        // On caret position change, highlight the corresponding symbol node
+        private void Caret_PositionChanged(object sender, CaretPositionChangedEventArgs e)
+        {
+            if (!e.NewPosition.Equals(e.OldPosition))
```

Comment:
Thanks for pointing that out! Correct me if I'm wrong but I believe that in this case, we don't need this to retrigger since the user would be selecting the exact same symbol node which is already highlighted.